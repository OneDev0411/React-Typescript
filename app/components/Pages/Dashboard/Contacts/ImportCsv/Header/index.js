import React from 'react'

const Header = ({ onClose }) => (
  <div className="profile__header">
    <h1 className="profile__header__title">
      <span>Import a CSV file</span>
    </h1>

    <div className="profile__header__close">
      <i className="fa fa-times" onClick={onClose} />
    </div>
  </div>
)

export default Header
