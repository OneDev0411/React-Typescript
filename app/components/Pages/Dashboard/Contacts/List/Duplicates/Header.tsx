import React, { useState } from 'react'
import { Button } from '@material-ui/core'
import pluralize from 'pluralize'

import PageHeader from 'components/PageHeader'
import { Trigger as MenuTrigger } from 'components/SlideMenu'

interface Props {
  listsLength: number
  isSideMenuOpen: boolean
  onSideMenuTriggerClick: () => void
  onMergeAllClick: () => Promise<void>
}

export default function Header({
  listsLength,
  isSideMenuOpen,
  onSideMenuTriggerClick,
  onMergeAllClick
}: Props) {
  const [isMerging, setIsMerging] = useState(false)

  const getMergeAllButtonCopy = () => {
    const possiblePluralizedList = pluralize('contacts list', listsLength, true)

    if (isMerging) {
      return `Merging ${possiblePluralizedList}`
    }

    return `Merge ${possiblePluralizedList}`
  }

  const handleMergeAll = async () => {
    setIsMerging(true)

    await onMergeAllClick()

    setIsMerging(false)
  }

  return (
    <PageHeader style={{ height: '5.0625rem' }}>
      <PageHeader.Title showBackButton={false}>
        <MenuTrigger isOpen={isSideMenuOpen} onClick={onSideMenuTriggerClick} />
        <PageHeader.Heading>Duplicate Contacts</PageHeader.Heading>
      </PageHeader.Title>
      {listsLength > 0 && (
        <PageHeader.Menu>
          <Button
            variant="contained"
            color="primary"
            disabled={isMerging}
            onClick={handleMergeAll}
          >
            {getMergeAllButtonCopy()}
          </Button>
        </PageHeader.Menu>
      )}
    </PageHeader>
  )
}
