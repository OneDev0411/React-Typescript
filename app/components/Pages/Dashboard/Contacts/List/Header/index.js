import React from 'react'
import { browserHistory } from 'react-router'

import Import from '../Import'
import PageHeader from '../../../../../../views/components/PageHeader'
import ActionButton from '../../../../../../views/components/Button/ActionButton'

export function Header(props) {
  return (
    <PageHeader title="Contacts" backButton={false}>
      <PageHeader.Menu>
        <Import userId={props.user.id} />
        <ActionButton
          style={{ padding: '0.75em' }}
          onClick={() => browserHistory.push('/dashboard/contacts/new')}
        >
          New Contact
        </ActionButton>
      </PageHeader.Menu>
    </PageHeader>
  )
}

export default Header
