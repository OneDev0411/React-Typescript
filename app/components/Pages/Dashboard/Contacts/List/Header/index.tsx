import { Grid } from '@material-ui/core'

import ContactsSegments from '../ContactsSegments'
import ContactFilters from '../Filters'
import { SortFields } from '../SortFields'
import { SyncedContacts as SyncedContactsTypes } from '../utils/get-synced-contacts'
import { ViewSwitcher } from '../ViewSwitcher'

export type ViewModeType = 'table' | 'board'

export interface Props {
  handleFilterChange: (newFilters: object, resetLoadedRanges: boolean) => void
  handleResetShortcutFilter: () => void
  filter: {
    show: boolean
  }
  tagListProps: {
    onClick: (props: unknown) => void
    isActive: boolean
  }
  savedListProps: {
    name: string
    associations: string
    getPredefinedLists: () => void
    onChange: (segment: unknown) => void
  }
  sortProps: {
    onChange: (item: unknown) => void
    currentOrder: string
    searchValue: string
  }
  onChangeView: (viewMode: ViewModeType) => void
  viewMode: ViewModeType
  contactCount: number
  activeSegment: any
  users: UUID[]
  syncedContacts: SyncedContactsTypes
}

export const Header = ({
  handleResetShortcutFilter,
  handleFilterChange,
  savedListProps,
  onChangeView,
  activeSegment,
  contactCount,
  tagListProps,
  sortProps,
  viewMode,
  filter,
  users
}: Props) => {
  return (
    <Grid direction="column" container>
      <Grid>
        <ContactsSegments
          activeSegment={activeSegment}
          handleResetShortcutFilter={handleResetShortcutFilter}
          handleFilterChange={handleFilterChange}
          savedListProps={savedListProps}
          tagListProps={tagListProps}
        />
      </Grid>
      <Grid container justifyContent="space-between" alignItems="center">
        <ContactFilters
          show={filter?.show}
          contactCount={contactCount}
          activeSegment={activeSegment}
          onFilterChange={() => handleFilterChange({}, true)}
          viewMode={viewMode}
          users={users}
        />
        <Grid>
          <SortFields {...sortProps} />
          <ViewSwitcher onChangeView={onChangeView} activeView={viewMode} />
        </Grid>
      </Grid>
    </Grid>
  )
}

export default Header
