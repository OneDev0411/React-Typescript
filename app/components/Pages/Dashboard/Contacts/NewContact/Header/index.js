import React from 'react'
import PageHeader from '../../../../../../views/components/PageHeader'

export default function Header() {
  return <PageHeader title="All Contacts" backUrl="/dashboard/contacts" />
}
