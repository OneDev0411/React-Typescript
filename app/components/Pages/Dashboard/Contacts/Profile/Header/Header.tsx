import React from 'react'
import { Box } from '@material-ui/core'

import Menu from './Menu/Menu'
import Catalog from './Catalog/Catalog'

interface Props {
  contact: IContact
}

export default function Header({ contact }: Props) {
  return (
    <Box p={3}>
      <Catalog contact={contact} />
      <Menu contact={contact} />
    </Box>
  )
}
