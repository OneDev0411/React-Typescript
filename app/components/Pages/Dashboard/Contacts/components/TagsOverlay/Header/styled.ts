import styled from 'styled-components'

import { primary } from 'views/utils/colors'

export const HeaderContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  height: 100%;
  width: 100%;

  svg {
    color: ${props => props.theme.palette.common.black};
    &:hover {
      color: ${primary};
    }
  }
`
