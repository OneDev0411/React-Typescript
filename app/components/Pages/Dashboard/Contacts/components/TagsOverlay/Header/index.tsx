import React from 'react'

import { Tooltip } from '@material-ui/core'
import { mdiCog, mdiClose } from '@mdi/js'
import { Link } from 'react-router'
import Flex from 'styled-flex-component'

import IconButton from 'components/Button/IconButton'
import { Title } from 'components/OverlayDrawer/Header/styled'
import { muiIconSizes } from 'components/SvgIcons/icon-sizes'
import { SvgIcon } from 'components/SvgIcons/SvgIcon'

import { HeaderContainer } from './styled'

interface Props {
  title: string
  onClose: () => any
}

export default function Header({ title, onClose }: Props) {
  return (
    <HeaderContainer>
      <Flex alignCenter>
        <Title
          style={{
            display: 'inline-block'
          }}
        >
          {title}
        </Title>
      </Flex>

      <Flex alignCenter>
        <Link
          style={{
            marginRight: '0.5rem',
            paddingTop: '0.3rem'
          }}
          to="/dashboard/account/manage-tags"
        >
          <Tooltip title="Manage tags" placement="bottom">
            <SvgIcon path={mdiCog} size={muiIconSizes.small} />
          </Tooltip>
        </Link>
        <IconButton
          type="button"
          isFit
          iconSize="large"
          inverse
          onClick={onClose}
        >
          <SvgIcon path={mdiClose} />
        </IconButton>
      </Flex>
    </HeaderContainer>
  )
}
