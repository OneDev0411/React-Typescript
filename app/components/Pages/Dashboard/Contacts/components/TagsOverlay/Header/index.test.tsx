import { mdiClose } from '@mdi/js'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Link } from 'react-router'

import IconButton from 'components/Button/IconButton'

import Header from '.'

describe('TagsOverlay header', () => {
  it('renders', () => {
    const wrapper = shallow(
      <Header title="manage tags here" onClose={() => null} />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // it('renders proper title', () => {
  //   const wrapper = shallow(
  //     <Header title="manage tags here" onClose={() => null} />
  //   )

  //   expect(wrapper.text()).toContain('manage tags here')
  // })

  it('renders link to tag management settings', () => {
    const wrapper = shallow(
      <Header title="manage tags here" onClose={() => null} />
    )

    expect(wrapper.find(Link)).toHaveLength(1)
  })

  it('renders close button', () => {
    const wrapper = shallow(
      <Header title="manage tags here" onClose={() => null} />
    )

    expect(wrapper.find(IconButton)).toHaveLength(1)
    expect(wrapper.find({ path: mdiClose })).toHaveLength(1)
  })
})
