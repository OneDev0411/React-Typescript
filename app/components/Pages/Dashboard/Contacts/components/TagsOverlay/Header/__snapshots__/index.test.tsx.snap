// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsOverlay header renders 1`] = `
<styled__HeaderContainer>
  <flex
    alignCenter={true}
  >
    <styled__Title
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      manage tags here
    </styled__Title>
  </flex>
  <flex
    alignCenter={true}
  >
    <Link
      onlyActiveOnIndex={false}
      style={
        Object {
          "marginRight": "0.5rem",
          "paddingTop": "0.3rem",
        }
      }
      to="/dashboard/account/manage-tags"
    >
      <_default
        caption="Manage your tags"
        placement="bottom"
      >
        <IconCog
          fillColor={null}
          fillColorRule="&&& path, &&& use, &&& g"
          size={null}
          sizes={Object {}}
          viewBox="0 0 24 24"
        >
          <path
            d="M22.5 9.5h-1.862a8.952 8.952 0 0 0-.763-1.84l1.317-1.316a1.5 1.5 0 0 0 0-2.122l-1.415-1.413a1.5 1.5 0 0 0-2.121 0l-1.318 1.317a8.906 8.906 0 0 0-1.838-.763V1.5A1.5 1.5 0 0 0 13 0h-2a1.5 1.5 0 0 0-1.5 1.5v1.863a8.969 8.969 0 0 0-1.839.763L6.343 2.809a1.5 1.5 0 0 0-2.121 0L2.807 4.223a1.5 1.5 0 0 0 0 2.122l1.318 1.317A8.878 8.878 0 0 0 3.362 9.5H1.5A1.5 1.5 0 0 0 0 11v2a1.5 1.5 0 0 0 1.5 1.5h1.862c.185.64.44 1.257.763 1.839l-1.319 1.319a1.5 1.5 0 0 0 0 2.121l1.414 1.414a1.545 1.545 0 0 0 2.122 0l1.318-1.317a8.98 8.98 0 0 0 1.84.764v1.86A1.5 1.5 0 0 0 11 24h2a1.5 1.5 0 0 0 1.5-1.5v-1.86a8.97 8.97 0 0 0 1.839-.763l1.318 1.317a1.5 1.5 0 0 0 2.121 0l1.414-1.414a1.5 1.5 0 0 0 0-2.121l-1.317-1.317c.322-.583.578-1.2.763-1.84H22.5A1.5 1.5 0 0 0 24 13v-2a1.5 1.5 0 0 0-1.5-1.5zM12 17a5 5 0 1 1 0-9.999A5 5 0 0 1 12 17z"
            key="key-0"
          />
        </IconCog>
      </_default>
    </Link>
    <IconButton
      appearance="icon"
      brandColor=""
      disabled={false}
      iconSize="large"
      inverse={true}
      isActive={false}
      isBlock={false}
      isFit={true}
      onClick={[Function]}
      size="medium"
      type="button"
    >
      <CloseIcon
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          key="key-0"
        />
      </CloseIcon>
    </IconButton>
  </flex>
</styled__HeaderContainer>
`;
